File: Android-SDK/phoneagent-sdk/src/main/java/com/autoglm/phoneagent/agent/AgentConfig.kt
package com.autoglm.phoneagent.agent

/**
 * Configuration for the AgentEngine.
 *
 * @property maxSteps Maximum number of steps to execute before stopping
 * @property stepDelay Delay between steps in milliseconds to allow UI to update
 * @property verbose Whether to log detailed information
 * @property autoConfirmSensitiveActions Whether to auto-confirm sensitive actions (default: false)
 * @property unknownActionThreshold Number of consecutive unknown actions before stopping (default: 3)
 */
data class AgentConfig(
    val maxSteps: Int = 50,
    val stepDelay: Long = 500,
    val verbose: Boolean = true,
    val autoConfirmSensitiveActions: Boolean = false,
    val unknownActionThreshold: Int = 3
) {
    companion object {
        /**
         * Default configuration with all default values.
         */
        val DEFAULT = AgentConfig()

        /**
         * Fast configuration with reduced delays and fewer steps for quick testing.
         */
        val FAST = AgentConfig(
            maxSteps = 20,
            stepDelay = 300,
            verbose = false
        )

        /**
         * Debug configuration with verbose logging and longer delays.
         */
        val DEBUG = AgentConfig(
            maxSteps = 100,
            stepDelay = 1000,
            verbose = true
        )

        /**
         * Production configuration optimized for production use.
         */
        val PRODUCTION = AgentConfig(
            maxSteps = 50,
            stepDelay = 500,
            verbose = false,
            autoConfirmSensitiveActions = false
        )
    }

    /**
     * Validate the configuration.
     * @throws IllegalArgumentException if configuration is invalid
     */
    fun validate() {
        require(maxSteps > 0) { "maxSteps must be greater than 0" }
        require(stepDelay >= 0) { "stepDelay must be non-negative" }
        require(unknownActionThreshold > 0) { "unknownActionThreshold must be greater than 0" }
    }
}

File: iOS-SDK/PhoneAgentDemo/SDK/Agent/AgentConfig.swift
//
//  AgentConfig.swift
//  PhoneAgentSDK
//
//  Created by AutoGLM on 2025-01-30.
//

import Foundation

/// Agent 配置
public struct AgentConfig {

    /// 最大执行步数
    public var maxSteps: Int

    /// 每步操作后的等待时间（秒）
    public var stepDelay: TimeInterval

    /// 是否打印详细日志
    public var verbose: Bool

    /// 是否自动处理敏感操作（如果为 false，需要设置回调）
    public var autoConfirmSensitiveActions: Bool

    public init(
        maxSteps: Int = 50,
        stepDelay: TimeInterval = 0.5,
        verbose: Bool = true,
        autoConfirmSensitiveActions: Bool = false
    ) {
        self.maxSteps = maxSteps
        self.stepDelay = stepDelay
        self.verbose = verbose
        self.autoConfirmSensitiveActions = autoConfirmSensitiveActions
    }
}

// MARK: - 默认配置
extension AgentConfig {
    /// 默认配置
    public static let `default` = AgentConfig()

    /// 快速测试配置（较少步数，较短延迟）
    public static let fast = AgentConfig(maxSteps: 20, stepDelay: 0.3)

    /// 详细调试配置
    public static let debug = AgentConfig(verbose: true)
}
